src/App.tsx

```diff
@@ -26,70 +28,116 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render to prepare goods
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
 
-  return visibleGoods;
-}
+  reverseGoods = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+  sortGoodsByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
 
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
-};
+  sortGoodsAlpha = () => {
+    this.setState({ sortType: SortType.ALPABET });
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  render() {
+    function getReorderedGoods(
+      goods: string[],
+      { sortType, isReversed }: ReorderOptions,
+    ) {
+      const visibleGoods = [...goods];
+
+      visibleGoods.sort((a, b) => {
+        switch (sortType) {
+          case SortType.LENGTH:
+            return a.length - b.length;
+
+          case SortType.ALPABET:
+            return a.localeCompare(b);
+
+          default:
+            return 0;
+        }
+      });
+
+      if (isReversed) {
+        visibleGoods.reverse();
+      }
+
+      return visibleGoods;
+    }
```

Comment:
Why you define your function inside render() method?

Reply:
the truth is that i dont know) thanks